airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/state/StateManagerFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.relationaldb.state;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factory class that creates {@link StateManager} instances based on the provided state.
+ */
+public class StateManagerFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(StateManagerFactory.class);
+
+  /**
+   * Private constructor to prevent direct instantiation.
+   */
+  private StateManagerFactory() {}
+
+  /**
+   * Creates a {@link StateManager} based on the provided state object and catalog.
+   *
+   * @param state The deserialized state.
+   * @param catalog The {@link ConfiguredAirbyteCatalog} for the connector that will utilize the state
+   *        manager.
+   * @param config The connector configuration.
+   * @return A newly created {@link StateManager} implementation based on the provided state.
+   */
+  public static StateManager createStateManager(final List<AirbyteStateMessage> state,
+                                                final ConfiguredAirbyteCatalog catalog,
+                                                final JsonNode config) {
+    if (state != null && !state.isEmpty()) {
+      final AirbyteStateMessage airbyteStateMessage = state.get(0);
+      if (isCdc(config) || airbyteStateMessage.getStateType() == AirbyteStateType.GLOBAL) {
+        LOGGER.info("Global state manager selected to manage state object with type {}.", airbyteStateMessage.getStateType());
+        return new GlobalStateManager(airbyteStateMessage, catalog);
+      } else if (airbyteStateMessage.getData() != null) {
```

Comment:
This condition is always going to be true since we populate the data field. I could not figure out how would we make sure this is going to be false for Postgres?

Reply:
@subodh1810 There is a bug here (now fixed).  I needed to add a check to make sure that the steam list is empty so that we know we are in the legacy state.